jwql/instrument_monitors/common_monitors/dark_monitor.py

```diff
@@ -0,0 +1,913 @@
+            # run
+            limits = ascii.read(THRESHOLDS_FILE)
+
+            # Use the current time as the end time for MAST query
+            self.query_end = Time.now().mjd
+
+            # Loop over all instruments
+            for instrument in JWST_INSTRUMENT_NAMES:
+                self.instrument = instrument
+
+                # Identify which database tables to use
+                self.identify_tables()
+
+                # Get a list of all possible apertures from pysiaf
+                possible_apertures = list(Siaf(instrument).apernames)
+                possible_apertures = [ap for ap in possible_apertures if ap not in apertures_to_skip]
+
+                for aperture in possible_apertures:
+                    # Find the appropriate threshold for the number of
+                    # new files needed
+                    match = aperture == limits['Aperture']
+                    new_dark_threshold = limits['Threshold'][match]
+
+                    # Locate the record of the most recent MAST search
+                    self.aperture = aperture
+                    self.query_start = self.most_recent_search()
+
+                    #logging.info('Working on aperture {} in {}'.format(aperture, instrument))
+                    #logging.info('Query times: {} {}'.format(self.query_start, self.query_end))
+
+                    # Query MAST using the aperture and the time of the
+                    # most recent previous search as the starting time
+                    new_entries = mast_query_darks(instrument, aperture, self.query_start, self.query_end)
+
+                    #logging.info('Aperture: {}, new entries: {}'.format(self.aperture, len(new_entries)))
+
+                    # Check to see if there are enough new files to meet
+                    # the monitor's signal-to-noise requirements
+                    if len(new_entries) >= new_dark_threshold:
+                        #logging.info("Sufficient new dark files found for {}, {} to run the dark monitor."
+                        #             .format(self.instrument, self.aperture))
+
+                        # Get full paths to the files
+                        new_filenames = [filesystem_path(file_entry['filename']) for file_entry in new_entries]
+
+                        # Set up directories for the copied data
+                        ensure_dir_exists(os.path.join(self.output_dir, 'data'))
+                        self.data_dir = os.path.join(self.output_dir,
+                                                     'data/{}_{}'.format(self.instrument.lower(),
+                                                                         self.aperture.lower()))
+                        ensure_dir_exists(self.data_dir)
+
+                        # Copy files from filesystem
+                        dark_files, not_copied = copy_files(new_filenames, self.data_dir)
+
+                        # Short-term fix: if some of the files from the query are not
```

Comment:
@bhilbert4 Yes, I think this code block should be removed.  If we encounter missing files in the filesystem, that is an issue beyond that of the dark monitor that we will have to resolve.

Reply:
Removed. Is there a reason to keep the `download_mast_data` function? Seems like it could potentially be useful to someone...maybe.

Reply:
@bhilbert4 Yeah we can keep that around for now.  It has a nice place in the `utils` module.